File: eco-core/core-plugin/src/main/java/com/willfp/ecomc/SchmoneyPlaceholder.java
package com.willfp.ecomc;

import com.willfp.eco.core.integrations.economy.EconomyManager;
import com.willfp.eco.core.integrations.placeholder.PlaceholderEntry;
import com.willfp.eco.core.integrations.placeholder.PlaceholderManager;
import com.willfp.eco.util.NumberUtils;
import lombok.experimental.UtilityClass;
import org.bukkit.Bukkit;
import org.bukkit.entity.Player;

import java.util.Map;
import java.util.UUID;
import java.util.concurrent.ConcurrentHashMap;

@UtilityClass
public class SchmoneyPlaceholder {
    private static final EcoMCPlugin PLUGIN = EcoMCPlugin.getInstance();
    private static final Map<UUID, Double> PREVIOUS_BALANCES = new ConcurrentHashMap<>();

    public void init() {
        PlaceholderManager.registerPlaceholder(new PlaceholderEntry(
                "money_change",
                player -> {
                    double prev = PREVIOUS_BALANCES.getOrDefault(player.getUniqueId(), 0.0);
                    if (prev == 0) {
                        return "";
                    }

                    double diff = EconomyManager.getBalance(player) - prev;
                    if (Math.abs(diff) > 0.01) {
                        if (diff > 0) {
                            return " §e+" + NumberUtils.format(diff) + "";
                        } else {
                            return " §e" + NumberUtils.format(diff) + "";
                        }
                    } else {
                        return "";
                    }
                }
        ));
    }

    public void createTheRunnable() {
        PLUGIN.getScheduler().runTimer(() -> {
            for (Player player : Bukkit.getOnlinePlayers()) {
                PREVIOUS_BALANCES.put(player.getUniqueId(), EconomyManager.getBalance(player));
            }
        }, 20, 60);
    }
}

File: eco-core/core-plugin/src/main/kotlin/com/willfp/ecomc/SchmoneyPlaceholder.kt
package com.willfp.ecomc

import com.willfp.eco.core.EcoPlugin
import com.willfp.eco.core.integrations.economy.EconomyManager
import com.willfp.eco.core.integrations.placeholder.PlaceholderManager
import com.willfp.eco.core.placeholder.PlayerPlaceholder
import com.willfp.eco.util.NumberUtils
import org.bukkit.Bukkit
import java.util.UUID
import java.util.concurrent.ConcurrentHashMap

object SchmoneyPlaceholder {
    private val PREVIOUS_BALANCES: MutableMap<UUID, Double> = ConcurrentHashMap<UUID, Double>()

    fun init() {
        PlaceholderManager.registerPlaceholder(
            PlayerPlaceholder(
                EcoMCPlugin.instance,
                "money_change"
            ) {
                val prev = PREVIOUS_BALANCES.getOrDefault(it.uniqueId, 0.0)
                if (prev == 0.0) {
                    return@PlayerPlaceholder ""
                }
                val diff: Double = EconomyManager.getBalance(it) - prev
                if (Math.abs(diff) > 0.01) {
                    if (diff > 0) {
                        return@PlayerPlaceholder " §e+" + NumberUtils.format(diff) + ""
                    } else {
                        return@PlayerPlaceholder " §e" + NumberUtils.format(diff) + ""
                    }
                } else {
                    return@PlayerPlaceholder ""
                }
            }
        )
    }

    fun createTheRunnable(plugin: EcoPlugin) {
        plugin.scheduler.runTimer({
            for (player in Bukkit.getOnlinePlayers()) {
                PREVIOUS_BALANCES[player.uniqueId] = EconomyManager.getBalance(player)
            }
        }, 20, 60)
    }
}